yt_dlp/postprocessor/modify_chapters.py

```diff
@@ -0,0 +1,326 @@
+            else:
+                self.report_warning(
+                    f'Skipping {self.pp_key()} since the video was already downloaded. '
+                    'Use --force-remove-chapters to remove the chapters anyway')
+                return [], info
+
+        concat_opts = self._make_concat_opts(cuts, duration)
+
+        def remove_chapters(file, is_sub):
+            return file, self.remove_chapters(file, cuts, concat_opts, self._force_keyframes and not is_sub)
+
+        in_out_files = [remove_chapters(info['filepath'], False)]
+        in_out_files.extend(remove_chapters(in_file, True) for in_file in self._get_supported_subs(info))
+
+        # Renaming should only happen after all files are processed
+        files_to_remove = []
+        for in_file, out_file in in_out_files:
+            uncut_file = prepend_extension(in_file, 'uncut')
+            if os.path.exists(uncut_file):
+                os.remove(uncut_file)
+            os.rename(in_file, uncut_file)
+            os.rename(out_file, in_file)
+            files_to_remove.append(uncut_file)
+
+        return files_to_remove, info
+
+    def _mark_chapters_to_remove(self, chapters, sponsor_chapters):
+        chapters = copy.deepcopy(chapters or [])
+        if self._remove_chapters_patterns:
+            warn_no_chapter_to_remove = True
+            if not chapters:
+                self.to_screen('Chapter information is unavailable')
+                warn_no_chapter_to_remove = False
+            for c in chapters:
+                if any(regex.search(c['title']) for regex in self._remove_chapters_patterns):
+                    c['remove'] = True
+                    warn_no_chapter_to_remove = False
+            if warn_no_chapter_to_remove:
+                self.to_screen('There are no chapters matching the regex')
+
+        sponsor_chapters = copy.deepcopy(sponsor_chapters or [])
+        if self._remove_sponsor_segments:
+            warn_no_chapter_to_remove = True
+            if not sponsor_chapters:
+                self.to_screen('SponsorBlock information is unavailable')
+                warn_no_chapter_to_remove = False
+            for c in sponsor_chapters:
+                if c['category'] in self._remove_sponsor_segments:
+                    c['remove'] = True
+                    warn_no_chapter_to_remove = False
+            if warn_no_chapter_to_remove:
+                self.to_screen('There are no matching SponsorBlock chapters')
+
+        return chapters, sponsor_chapters
+
+    def _get_real_video_duration(self, filename):
+        try:
+            duration = float_or_none(
+                traverse_obj(self.get_metadata_object(filename), ('format', 'duration')))
+            if duration is None:
+                raise PostProcessingError('ffprobe returned empty duration')
+            return duration
+        except PostProcessingError as err:
+            raise PostProcessingError(f'Cannot determine the video duration; {err}')
```

Comment:
I haven't thought that Python would print `During handling of the above exception, another exception occurred`, I thought it's gonna be lost. Anyway, exceptions are usually chained with `raise ... from e`, which will produce `The above exception was the direct cause of the following exception`.

Message of an inner exception is obscure because it's a test exception, not a real one. The real exception would have a better description.